### PR TITLE
test that compiler accounts for bad ro fidelity

### PR DESCRIPTION
**Pull request by erichulburd**

### CODE REVIEW

File: tests/compiler-hook-tests.lisp [hunk 1]
```diff
@@ -555,3 +555,14 @@ CCNOT 0 1 2")
         (signals quil::illegal-qubits-used-in-preserved-block-error (compiler-hook bad-preserve chip-bad))
         (signals quil::illegal-qubits-used-in-preserved-block-error (compiler-hook bad-preserve-measure chip-bad))
         (signals quil::illegal-qubits-used-in-preserved-block-error (compiler-hook bad-preserve-reset chip-bad))))))
+
+(deftest test-compiler-accounts-for-readout-fidelity
```
Comment (erichulburd) [inline tests/compiler-hook-tests.lisp:559]: Figured `compiler-hook` is the easiest high level call to make to test compilation from raw quil string, thus I've added to `compiler-hook-tests.lisp`. LMK if there is a better home for this test.

File: tests/qpu-test-files/Aspen-4-2Q-B.qpu [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (erichulburd) [inline tests/qpu-test-files/Aspen-4-2Q-B.qpu:1]: Figured I would just throw this in with the other QPU fixtures. Alternatively, I could just inline this as a string, or give it a more descriptive name, such as `Aspen-4-2Q-ro-fidelity-discrepancy.qpu`.

File: tests/qpu-test-files/Aspen-4-2Q-B.qpu [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "metadata": {
+    "name": "Aspen-4-2Q-B",
+    "version": 0.0,
+    "timestamp": "2019-07-31 22:59:15.050401"
+  },
+  "isa": {
+    "1Q": {
+      "1": {
```
Comment (erichulburd) [inline tests/qpu-test-files/Aspen-4-2Q-B.qpu:9]: I originally had `"gates"` array here with RZ operators defined with `fRO` defined, but I kept getting 

```
User program used too many qubits: 1 used and 0 available in the largest connected component
```

**Comment by notmgsk** [comment]
Closing in favor of #699 (which cherry picks this).

**Pull request closed without merging**